spec/features/5_photos_spec.rb

```diff
@@ -1,120 +1,126 @@
 require "rails_helper"
 
-show_tests_in_browser = false
-do_not_show_tests_in_browser = false
+feature "Photos:" do
+
+  context "photos show page" do
+    before(:each) do
+      @user_1 = FactoryGirl.create(:user, :username => "1", :email => "1@m.com")
+      @user_2 = FactoryGirl.create(:user, :username => "2", :email => "2@m.com")
+      @user_3 = FactoryGirl.create(:user, :username => "Three", :email => "three@m.com")
+      @photo_1 = FactoryGirl.create(:photo, :user_id => @user_1.id)
+      @photo_2 = FactoryGirl.create(:photo, :user_id => @user_2.id)
+      @photo_3 = FactoryGirl.create(:photo, :user_id => @user_3.id)
+    end
 
-feature "Photos:", js: do_not_show_tests_in_browser do
+    scenario "only show edit link for signed-in user's photos", points: 1 do
+      login_as(@user_2, :scope => :user)
+      visit "/photos"
 
-  scenario "only show edit/delete for signed-in user's photos", points: 2 do
-    user_1 = FactoryGirl.create(:user, :username => "1", :email => "1@m.com")
-    user_2 = FactoryGirl.create(:user, :username => "2", :email => "2@m.com")
-    photo_1 = FactoryGirl.create(:photo, :user_id => user_1.id)
-    photo_2 = FactoryGirl.create(:photo, :user_id => user_2.id)
-    login_as(user_2, :scope => :user)
+      @user_2.photos.each do |photo|
+        expect(page).to have_link(nil, href: "/photos/#{photo.id}/edit")
+      end
 
-    visit "/photos"
+      @user_1.photos.each do |photo|
+        expect(page).not_to have_link(nil, href: "/photos/#{photo.id}/edit")
+      end
+    end
 
-    expect(page).not_to have_link(nil, href: "/photos/#{photo_1.id}/edit")
-    expect(page).not_to have_link(nil, href: "/delete_photo/#{photo_1.id}")
-    expect(page).to have_link(nil, href: "/photos/#{photo_2.id}/edit")
-    expect(page).to have_link(nil, href: "/delete_photo/#{photo_2.id}")
-  end
+    scenario "only show delete link for signed-in user's photos", points: 1 do
+      login_as(@user_2, :scope => :user)
+      visit "/photos"
 
-  scenario "in /photos, Bootstrap panels used", points: 1 do
-    user = FactoryGirl.create(:user)
-    photo = FactoryGirl.create(:photo, :user_id => user.id)
-    login_as(user, :scope => :user)
+      @user_2.photos.each do |photo|
+        expect(page).to have_link(nil, href: "/delete_photo/#{photo.id}")
+      end
 
-    visit "/photos"
+      @user_1.photos.each do |photo|
+        expect(page).not_to have_link(nil, href: "/delete_photo/#{photo.id}")
+      end
+    end
 
-    expect(page).to have_css(".panel")
-  end
+    scenario "in /photos, Bootstrap panels used", points: 1 do
```

Comment:
Do any later tests depend on `.panel` being present? If so, we should remove this test; I don't want to test styling at this stage. Not sure why this was originally in here.